core/modules/catalog/store/tax/actions.ts

```diff
@@ -5,27 +5,67 @@ import SearchQuery from '@vue-storefront/core/lib/search/searchQuery'
 import RootState from '@vue-storefront/core/types/RootState'
 import TaxState from '../../types/TaxState'
 import { Logger } from '@vue-storefront/core/lib/logger'
+import { StorageManager } from '@vue-storefront/core/lib/storage-manager'
+import { entityKeyName } from '@vue-storefront/core/lib/store/entities'
+import config from 'config'
+import { currentStoreView } from '@vue-storefront/core/lib/multistore'
+import { calculateProductTax } from '@vue-storefront/core/modules/catalog/helpers/taxCalc'
+import { doPlatformPricesSync } from '@vue-storefront/core/modules/catalog/helpers'
 
 const actions: ActionTree<TaxState, RootState> = {
-  /**
-   * Load the tax rules
-   */
-  list (context, { entityType = 'taxrule' }) {
-    if (context.state.rules.length > 0) {
+  async list ({ state, commit, dispatch }, { entityType = 'taxrule' }) {
+    if (state.rules.length > 0) {
       Logger.info('Tax rules served from local memory', 'tax')()
-      return new Promise((resolve, reject) => {
-        resolve({ items: context.state.rules })
-      })
-    } else {
-      const searchQuery = new SearchQuery()
-      return quickSearchByQuery({ query: searchQuery, entityType }).then((resp) => {
-        context.commit(types.TAX_UPDATE_RULES, resp)
-        return resp
+      return { items: state.rules }
+    }
+
+    const resp = await quickSearchByQuery({ query: new SearchQuery(), entityType })
+    dispatch('storeToRulesCache', { items: resp.items })
+    commit(types.TAX_UPDATE_RULES, resp)
+
+    return resp
+  },
+  storeToRulesCache (context, { items }) {
+    const cache = StorageManager.get('elasticCache')
+
+    for (let tc of items) {
+      const cacheKey = entityKeyName('tc', tc.id)
+      cache.setItem(cacheKey, tc).catch((err) => {
+        Logger.error('Cannot store cache for ' + cacheKey + ', ' + err)()
       })
     }
   },
-  single (context, { productTaxClassId }) {
-    return context.state.rules.find((e) => { return e.product_tax_class_ids.indexOf(parseInt(productTaxClassId)) >= 0 })
+  single ({ getters }, { productTaxClassId }) {
+    return getters.getRules.find((e) =>
```

Comment:
When we are on it don't we need to add the `userGroupId` here too?

I am not 100% sure but I would think so. 
Maybe someone can give me a hint for what this is? So we know after we have clear for what it is if we need the `userGroupId` logic here too.